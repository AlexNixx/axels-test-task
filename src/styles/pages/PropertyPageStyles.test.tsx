import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {
    StyledButton,
    StyledContainer,
    StyledImage
} from './PropertyPageStyles';

describe('Test PropertyPageStyles', () => {
    test('Check StyledContainer styles', () => {
        const tree = renderer.create(<StyledContainer />).toJSON();
        expect(tree).toHaveStyleRule('padding-top', '2rem');
        expect(tree).toHaveStyleRule('padding-bottom', '2rem');
        expect(tree).toMatchSnapshot();
    });

    test('Check StyledImage styles', () => {
        const tree = renderer.create(<StyledImage />).toJSON();
        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('max-height', '30rem');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toMatchSnapshot();
    });

    test('Check StyledButton styles', () => {
        const tree = renderer.create(<StyledButton />).toJSON();
        expect(tree).toHaveStyleRule('width', '10rem');
        expect(tree).toHaveStyleRule('margin-bottom', '2rem');
        expect(tree).toMatchSnapshot();
    });
});
