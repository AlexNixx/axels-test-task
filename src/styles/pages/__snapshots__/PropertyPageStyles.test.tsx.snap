// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PropertyPageStyles Check StyledButton styles 1`] = `
.c0 {
  width: 10rem;
  margin-bottom: 2rem;
}

<button
  className="c0 btn btn-primary"
  disabled={false}
  type="button"
/>
`;

exports[`Test PropertyPageStyles Check StyledContainer styles 1`] = `
.c0 {
  padding-top: 2rem;
  padding-bottom: 2rem;
}

<div
  className="c0 container"
/>
`;

exports[`Test PropertyPageStyles Check StyledImage styles 1`] = `
.c0 {
  width: 100%;
  max-height: 30rem;
  overflow: hidden;
}

<img
  className="c0"
/>
`;
